vulture_os/toolkit/api_parser/beyondtrust_pra/beyondtrust_pra.py

```diff
@@ -0,0 +1,278 @@
+                events = [events]
+
+            for event in events:
+                event['team'] = {
+                    "name": log['@name'],
+                    "id": log['@id'],
+                }
+                formated_logs.append(event)
+
+        last_datetime = [x['@timestamp'] for x in formated_logs]
+        # ISO8601 timestamps are sortable as strings
+        last_datetime = max(last_datetime, default=None)
+        return formated_logs, last_datetime
+
+    @staticmethod
+    def format_access_session_logs(logs):
+        formated_logs = []
+        sessions_logs = logs['session_list'].get('session', [])
+        if isinstance(sessions_logs, list):
+            for log in sessions_logs:
+                del log['session_details']  # Avoid too long session log
+                log["@timestamp"] = log["start_time"]["@timestamp"]
+                formated_logs.append(log)
+        elif isinstance(sessions_logs, dict):
+            del sessions_logs['session_details']  # Avoid too long session log
+            sessions_logs["@timestamp"] = sessions_logs["start_time"]["@timestamp"]
+            formated_logs.append(sessions_logs)
+        last_datetime = [x['@timestamp'] for x in formated_logs]
+        # ISO8601 timestamps are sortable as strings
+        last_datetime = max(last_datetime, default=None)
+        return formated_logs, last_datetime
+
+    @staticmethod
+    def format_vault_account_activity_list_logs(logs):
+        formated_logs = logs['vault_account_activity_list'].get('vault_account_activity', [])
+        if isinstance(formated_logs, dict):
+            formated_logs = [formated_logs]
+        last_datetime = [x['@timestamp'] for x in formated_logs]
+        # ISO8601 timestamps are sortable as strings
+        last_datetime = max(last_datetime, default=None)
+        return formated_logs, last_datetime
+
+    def get_logs(self, resource: str, requested_type: str, since: object):
+        url = f"{self.beyondtrust_pra_host}/api/{resource}"
+
+        parameters = {
+            'start_time': int(since.timestamp()),
+            'duration': self.DURATION,
+            'generate_report': requested_type,
+        }
+
+        res = self._execute_query(url, query=parameters)
+
+        logs = xmltodict.parse(res.text)
+
+        if error := logs.get('error'):
+            logger.info(f"[{__parser__}]:get_logs: Error while getting '{requested_type}' logs : {error.get('#text')}", extra={'frontend': str(self.frontend)})
+            return [], None
+
+        # Return only list of events
+        if resource == 'reporting' and requested_type == "Team":
+            return self.format_team_logs(logs)
+        elif resource == 'reporting' and requested_type == "AccessSession":
+            return self.format_access_session_logs(logs)
+        elif resource == 'reporting' and requested_type == "VaultAccountActivity":
+            return self.format_vault_account_activity_list_logs(logs)
+        else:
```

Comment:
it's not clear that 2 parameters are always returned from those formatting functions (formatted_logs and last_logtime), and the last_logtime calculation could be done only once in this main function (instead of each formatting function)
In addition, calculating a last_logtime is not something that should be done by a function called "format_*()"!